source/usage-examples/findOne.txt

```diff
@@ -3,3 +3,93 @@ Find a Document
+with the minimum or maximum value as specified in your sort criteria. You can
+also specify a **projection**, which configures the fields in the returned
+document. You can specify additional options by passing the ``FindOne()`` method
+a pointer to an instance of a ``FindOneOptions`` value. You can configure a
+``FindOneOptions`` value by using setter methods such as the
+``*FindOneOptions.SetProjection()`` method.
+
+..
+   return
+
+If the query filter you specified in your ``FindOne()`` method call matches one
+or more documents in the collection, it returns the first matched result. The
+return type ``SingleResult`` represents a single document returned from a query
+to a MongoDB instance. You can call the ``*SingleResult.Decode()`` method
+to unmarshall a ``SingleResult`` value into a variable. Pass the ``Decode()`` 
+method a pointer to the variable that you would like to contain your returned
+document. 
+
+..
+   error 
+
+The ``Decode()`` method will return the error value of your ``FindOne()`` method
+call. 
+
+- If a document matched your query, and there are no errors retrieving the
+  document, the ``Decode()`` method will return ``nil``. 
+- If no documents matched your query, the ``Decode()`` method will return an
+  ``ErrNoDocuments`` error. 
+- If there was an error retrieving your document, the ``Decode()`` method will
+  return the error. 
+
+Example
+-------
+
+The following code example retrieves a single document from the ``movies``
+collection. It uses the following arguments for the ``FindOne()`` method:
+
+- An empty **Context** instance returned from the ``context.TODO()`` function.
+- A **query document** that only matches movies with the title exactly matching
+  the text ``"The Room"``.
+- A **configuration** that specifies a **sort** on matched documents in
+  descending order by rating, such that the retrieved document has the highest
+  rating. Our configuration also specifies a **projection** for our results so
+  that our MongoDB instance only sends the ``title`` and ``imdb`` fields of
+  our retrieved document.
+
+.. include:: /includes/connect-guide-note.rst
+
+.. literalinclude:: findOne.go
+   :language: go
+
+If you run the example above, you should see a result that looks like this:
+
+.. literalinclude:: findOne.go.txt
+   :language: none
+
+For more information on the methods, interfaces, structs, and functions
+mentioned on this page, see the following API documentation:
+
+- :go-api:`*Collection.FindOne() </mongo#Collection.FindOne>`
+- `Context <https://pkg.go.dev/context#Context>`__
+- :go-api:`FindOneOptions </mongo/options#FindOneOptions>`
+- :go-api:`*FindOneOptions.SetProjection() </mongo/options#FindOneOptions.SetProjection>`
+- :go-api:`options.FindOne() </mongo/options#FindOne>`
+- :go-api:`SingleResult </mongo#SingleResult>`
+- :go-api:`*SingleResult.Decode() </mongo#SingleResult.Decode>`
+- `context.TODO() <https://pkg.go.dev/context#TODO>`__
```

Comment:
Since you've already included the link to the Context package, I wonder if just one of these links is sufficient.

Reply:
Thank you for the note. I agree it is a little bit awkward to have both links, but I am not sure it is redundant as before I am linking to the `Context` interface type and later I am linking to the `TODO()` function in the `context` package. I like linking to both, but I could see removing the `context.TODO()` function from the list as also being find.

I'd prefer to leave it in, but if you feel strongly about taking it out let me know and I will remove it.